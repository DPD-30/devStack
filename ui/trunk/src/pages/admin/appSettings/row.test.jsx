import React from 'react'
import { renderToString } from 'react-dom/server'
import { mount } from 'enzyme'

import PagesAppSettingsRow from './row.jsx'

jest.unmock('./row.jsx')

const props = {
    name: 'mockname',
    id: 'mockid',
    value: ''
}

describe('PagesAppSettingsRow', () => {
    it('should match snapshot', () => {
        const comp = renderToString(<PagesAppSettingsRow {...props} />)

        expect(comp).toMatchSnapshot()
    })

    it ('handles normal flow to update', () => {
        const comp = mount(<PagesAppSettingsRow {...props} />)
        const inputField = comp.find('input')
        const btnUpdate = comp.find('button.btn.btn-secondary')

        inputField.simulate('change', { target: { name: 'value', value: '123' }})

        jest.runAllTimers()

        expect(comp.instance().state.value).toBe('123')

        btnUpdate.simulate('click')

        jest.runAllTimers()

        expect(inputField.getDOMNode()).toHaveProperty('disabled')
        expect(btnUpdate.getDOMNode()).toHaveProperty('disabled')
    })

    it ('handles normal flow to delete', () => {
        const comp = mount(<PagesAppSettingsRow {...props} />)
        const btnDelete = comp.find('button.btn.btn-warning')
        
        btnDelete.simulate('click')

        jest.runAllTimers()

        expect(comp.instance().state.isDeleting).toBe(true)
    })
})
