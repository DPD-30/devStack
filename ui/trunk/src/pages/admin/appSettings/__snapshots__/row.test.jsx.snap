// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesAppSettingsRow should match snapshot 1`] = `
<tr data-reactroot>
  <td>
    mockname
  </td>
  <td>
    <input type="text"
           name="value"
           value
    >
  </td>
  <td>
    <button type="button"
            disabled
            class="btn btn-secondary disabled"
    >
      Save
    </button>
  </td>
  <td>
    <button type="button"
            class="btn btn-warning"
    >
      Delete
    </button>
  </td>
</tr>
`;
